hr_timesheet_begin_end_no_end/tests/test_timesheet_begin_end_no_end.py

```diff
@@ -0,0 +1,49 @@
+# SPDX-FileCopyrightText: 2024 Coop IT Easy SC
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+from odoo import exceptions, fields
+from odoo.tests.common import TransactionCase
+
+
+class TestBeginEndNoEnd(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.analytic = cls.env.ref("analytic.analytic_administratif")
+        cls.user = cls.env.ref("base.user_root")
+
+    def base_line(self):
+        return {
+            "name": "test",
+            "date": fields.Date.today(),
+            "time_start": 10.0,
+            "time_stop": 12.0,
+            "user_id": self.user.id,
+            "unit_amount": 2.0,
+            "account_id": self.analytic.id,
+            "amount": -60.0,
+        }
+
+    def test_check_end_is_zero(self):
+        line = self.base_line()
+        line.update({"time_stop": 0, "unit_amount": 0})
+        # No error.
+        self.env["account.analytic.line"].create(line)
+
+    def test_check_end_is_not_zero(self):
+        line = self.base_line()
+        line.update({"time_stop": 1, "unit_amount": 0})
+        with self.assertRaises(exceptions.ValidationError):
+            self.env["account.analytic.line"].create(line)
+
```

Comment:
it would be good to add tests that (try to) change an existing line and check whether it also works correctly.